griptape/tasks/image_variation_task.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from typing import Callable
+
+from attr import define, field
+
+from griptape.artifacts import ImageArtifact, TextArtifact
+from griptape.tasks import BaseImageGenerationTask, BaseTask
+from griptape.utils import J2
+
+
+@define
+class ImageVariationTask(BaseImageGenerationTask):
+    """A task that generates a variation of an image using a prompt. Accepts a text prompt and image as
+    input in one of the following formats:
+    - tuple of (template string, ImageArtifact)
+    - tuple of (TextArtifact, ImageArtifact)
+    - Callable that returns a tuple of (TextArtifact, ImageArtifact)
+
+    Attributes:
+        image_generation_engine: The engine used to generate the image.
+        negative_rulesets: List of negatively-weighted rulesets applied to the text prompt, if supported by the driver.
+        negative_rules: List of negatively-weighted rules applied to the text prompt, if supported by the driver.
+        output_dir: If provided, the generated image will be written to disk in output_dir.
+        output_file: If provided, the generated image will be written to disk as output_file.
+    """
+
+    _input: tuple[str, ImageArtifact] | [TextArtifact, ImageArtifact] | Callable[
+        [BaseTask], tuple[TextArtifact, ImageArtifact]
+    ] = field(default=None, kw_only=True)
+
+    @property
+    def input(self) -> (TextArtifact, ImageArtifact):
+        if (
+            isinstance(self._input, tuple)
+            and isinstance(self._input[0], TextArtifact)
+            and isinstance(self._input[1], ImageArtifact)
+        ):
+            return self._input
+        if (
+            isinstance(self._input, tuple)
+            and isinstance(self._input[0], str)
+            and isinstance(self._input[1], ImageArtifact)
+        ):
+            return TextArtifact(J2().render_from_string(self._input[0], **self.full_context)), self._input[1]
+        elif isinstance(self._input, Callable):
+            return self._input(self)
+        else:
+            raise ValueError("Input must be a tuple of (text, image) or a callable that returns such a tuple.")
```

Comment:
Verbose, but @collindutter this aligns with your previous changes in #494. Image generation tasks support typed inputs, not just template text inputs with filepath configurations.

This means the ImageVariationTask, for example, accepts:
- template string + image artifact (str, ImageArtifact)
- text artifact + image artifact
- Callable returning text + image artifacts

Reply:
Added potential simplification on other class.